### PR TITLE
Add Encryption functions with different parameters.

### PR DESCRIPTION
**Pull request by mattabullock**
Adding these functions to be used in Encryption.cpp in Server-Expensify so we can avoid converting `unsigned char*` to string and back to `unsigned char*`.

Tested with the following PRs: https://github.com/Expensify/Server-Expensify/pull/2442, https://github.com/Expensify/Web-Expensify/pull/20077, and https://github.com/Expensify/Web-Secure/pull/729

### CODE REVIEW

File: libstuff/libstuff.cpp [hunk 1]
```diff
@@ -1885,28 +1885,29 @@ string SAESEncrypt(const string& buffer, const string& ivStr, const string& key)
...
     encryptedBuffer.resize(paddedBuffer.size());
-    mbedtls_aes_crypt_cbc(&ctx, MBEDTLS_AES_ENCRYPT, (int)paddedBuffer.size(), iv, (unsigned char*)paddedBuffer.c_str(),
-                          (unsigned char*)encryptedBuffer.c_str());
+    mbedtls_aes_crypt_cbc(&ctx, MBEDTLS_AES_ENCRYPT, (int)paddedBuffer.size(), iv, (unsigned char*)paddedBuffer.c_str(), (unsigned char*)encryptedBuffer.c_str());
     return encryptedBuffer;
 }
 
+string SAESEncrypt(const string& buffer, const string& ivStr, const string& key) {
+    unsigned char iv[SAES_IV_SIZE];
+    memcpy(iv, ivStr.c_str(), SAES_IV_SIZE);
```
Comment (tylerkaraszewski) [inline libstuff/libstuff.cpp]: Verify that `ivStr` is big enough for this.

File: libstuff/libstuff.cpp [hunk 2]
```diff
@@ -1922,6 +1923,12 @@ string SAESDecrypt(const string& buffer, const string& ivStr, const string& key)
     return decryptedBuffer;
 }
 
+string SAESDecrypt(const string& buffer, const string& ivStr, const string& key) {
+    unsigned char iv[SAES_IV_SIZE];
+    memcpy(iv, ivStr.c_str(), SAES_IV_SIZE);
```
Comment (tylerkaraszewski) [inline libstuff/libstuff.cpp]: verify `ivStr` is big enough.

File: libstuff/libstuff.cpp [hunk 3]
```diff
@@ -2097,31 +2104,41 @@ string SHashSHA1(const string& buffer) {
...
     string out;
     out.resize(olen - 1); // -1 because trailing 0 is implied
-    mbedtls_base64_encode((unsigned char*)&out[0], olen, &olen, (unsigned char*)buffer.c_str(), buffer.size());
+    mbedtls_base64_encode((unsigned char*)&out[0], olen, &olen, buffer, size);
     return out;
 }
 
+string SEncodeBase64(const string& bufferString) {
+    unsigned char* buffer = (unsigned char*)bufferString.c_str();
+    return SEncodeBase64(buffer, bufferString.size());
```
Comment (tylerkaraszewski) [inline libstuff/libstuff.cpp]: Maybe:
```
return SEncodeBase64((unsigned char*)bufferString.c_str(), bufferString.size());
```

File: libstuff/libstuff.cpp [hunk 4]
```diff
@@ -2097,31 +2104,41 @@ string SHashSHA1(const string& buffer) {
...
     string out;
     out.resize(olen);
-    mbedtls_base64_decode((unsigned char*)&out[0], olen, &olen, (unsigned char*)buffer.c_str(), buffer.size());
+    mbedtls_base64_decode((unsigned char*)&out[0], olen, &olen, buffer, size);
     return out;
 }
 
+string SDecodeBase64(const string& bufferString) {
+    unsigned char* buffer = (unsigned char*)bufferString.c_str();
+    return SDecodeBase64(buffer, bufferString.size());
```
Comment (tylerkaraszewski) [inline libstuff/libstuff.cpp]: Same suggestion as `SEncode...`

**Comment by mattabullock** [comment]
updated.

**Comment by mattabullock** [comment]
bump!

**Pull request merged**